src/seidr.py

```diff
@@ -2,25 +2,55 @@
 import numpy as np
 from scipy.integrate import odeint
 import matplotlib.pyplot as plt
+import pandas as pd
+
+OUTPUT_DIRECTORY_PATH = "../output"
 
 class ModelResultsNotCalculatedError(Exception):
     def __init__(self, *args: object) -> None:
         super().__init__("you need to call the calculate method first",*args)
 
 class SEIDR:
-    def __init__(self,N, beta, gamma, mi, lambda_, epsilon, alpha, no_of_days) -> None:
-        self.N = N
+    def __init__(self, 
+                S = 9900,
+                E = 100, 
+                I = 0, 
+                D = 0, 
+                R = 0, 
+                beta = 0.5, 
+                gamma = 0.1, 
+                mi = 0.0002, 
+                lambda_ = 0.3, 
+                epsilon = 0.2, 
+                alpha = 0.01, 
+                no_of_days = 100):
+
+        # Susceptible
+        self.S0 = S
+        # exposed
+        self.E0 = E
+        # infected
+        self.I0 = I
+        # dead
+        self.D0 = D
+        # recovered
+        self.R0 = R
+        # Contact rate
         self.beta = beta
+        # mean recovery rate, (in 1/days).
         self.gamma = gamma
+        # natural death rate
         self.mi = mi
+        # natural birth rate
         self.lambda_ = lambda_
+        # exposed to infectious
         self.epsilon = epsilon
+        # fatality rate
         self.alpha = alpha
 
         self.no_of_days = no_of_days
         self.t = np.linspace(0, no_of_days, no_of_days+1)
 
-        self.results = None
```

Comment:
without this, if we call save_results before calculate, it crashes with a generic AttributeError instead of our custom Error. Either we keep this, or we throw out lines 96 and 97, imo better to keep this.